### DISCUSSION TITLE
Joint inversion of ERT and Travertine for velocity

### DISCUSSION THREAD

**Discussion by pouryaalidoust** [gimli-org/pyGIMLi#541] [Q&A]
Hi All,

I want to jointly invert the velocity model using my ERT and travel time field dataset. I read the petrophysical joint inversion page on the website but it inverts for saturation. How can I define the relationship between these two for inverting for underlying velocity? in the example, we need to define the relationship between each parameter and saturation and then introduce it in the joint inversion snippet. However, in the case of inverting  for example VS, What should we do?

**Comment by halbmy** [comment]
You need to transform your intrinsic parameters (resistivity and velocity) into a common one, which could be saturation or porosity (assuming full saturation), but of course you can also transform velocity into resistivity or the other way round. Either by some porosity model or by some experimental correlation. What you need is to define such a transformation (like transFwdArchieS or transFwdWyllieS in the example). 

The code (see API doc under physics-petro) looks like:
```
def transFwdWyllieS(phi, vm=4000, vw=1600, va=330):
    return pg.trans.TransLin((1/vw-1./va)*phi, (1-phi)/vm+phi/va)
```

and shows that it needs to be a transformation class. I just see that these are not documented at all, so we should create an example for it. In https://github.com/gimli-org/gimli/blob/master/pygimli/core/trans.py there is an example using a symlog transformation:
```
class TransSymLog(pgcore.RTrans):
    """Transformation using a bilogarithmic scaling."""

    def __init__(self, tol=1e-12):
        """Forward transformation."""
        super().__init__()
        self.tol = tol

    def trans(self, x):
        """Forward transformation."""
        return pgcore.log(1 + np.abs(x) / self.tol) * np.sign(x)

    def invTrans(self, y):
        """Inverse transformation."""
        return (pgcore.exp(np.abs(y)) - 1.) * self.tol * np.sign(y)

    def deriv(self, x):
        """Derivative of the transformation."""
        return 1. / (np.abs(x) / self.tol + 1) / self.tol
```
